autoload/ale/handlers.vim

```diff
@@ -116,6 +114,45 @@ function! ale#handlers#HandlePEP8Format(buffer, lines) abort
     return l:output
 endfunction
 
+
+function! ale#handlers#HandleFlake8Format(buffer, lines) abort
+    " Matches patterns line the following:
+    "
+    " stdin:6:6: E111 indentation is not a multiple of four
+    "
+    " Uses g:ale_python_flake8_error_codes to decide the severity of the
+    " message
+    let l:pattern = '^' . s:path_pattern . ':\(\d\+\):\?\(\d\+\)\?: \(\([[:alpha:]]\)[[:alnum:]]\+\) \(.*\)$'
+    let l:output = []
+
+    for l:line in a:lines
+        let l:match = matchlist(l:line, l:pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let l:code = l:match[3]
+        if (l:code ==# 'W291' || l:code ==# 'W293')
+                    \ && !g:ale_warn_about_trailing_whitespace
+            " Skip warnings for trailing whitespace if the option is off.
+            continue
+        endif
+
+        call add(l:output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': l:match[1] + 0,
+        \   'vcol': 0,
+        \   'col': l:match[2] + 0,
+        \   'text': l:code . ': ' . l:match[5],
+        \   'type': index(g:ale_python_flake8_error_codes , l:code) == -1 ? 'W' : 'E',
```

Comment:
How does your solution improve #430? Wouldn't it result in the same output that smkent was complaining about?

It's also a solution that I'm personally not interested in at all. To get what I would consider sensible behavior (errors for errors, warnings for style issues) in this situation, I would have to list [all of pycodestyle's error codes](https://pycodestyle.readthedocs.io/en/latest/intro.html#error-codes) in my vimrc.

> The flake8 errors and warnings are a superset for pep8/pycodestyle. So it makes sense to write these changes into the PEP8 handler, and use the same settings for both. It shouldn't cause any issues.

What about [ansible](https://github.com/w0rp/ale/blob/ecbb27680563a50d4dd981f968d659ef20bfbe30/ale_linters/ansible/ansible-lint.vim#L8) and [pylint](https://github.com/w0rp/ale/blob/e97dada261c4a69f43c5a6c34d349ad6246fe34c/ale_linters/python/pylint.vim)?